### PR TITLE
horizontal fixed layout added (each measure has the same size)

### PR DESCRIPTION
**Pull request by alexpavlov96**
Resolves: *(direct link to the issue)*

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [ ] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Seems this would fix a pretty frequent request, see https://musescore.org/en/node/313555.
If so maybe change the commit title to start with "Fix #313555: "?

File: src/engraving/libmscore/measure.cpp [hunk 1]
```diff
@@ -4823,4 +4824,238 @@ void Measure::computeMinWidth()
...
+    x += segment->extraLeadingSpace().val() * spatium();
+
+    return x;
+}
+
+static Segment* findFirstEnabledSegment(Measure* measure)
+{
+    Segment* current = nullptr;
+    for (current = measure->first(); current && !current->enabled(); current = current->next()) {
+    }
```
Comment (Jojo-Schmitz) [inline src/engraving/libmscore/measure.cpp:4859]: Empty `for` loop?

**Comment by Jojo-Schmitz** [comment]
Isn't there any UI to switch this on/off?

**Comment by alexpavlov96** [comment]
> Seems this would fix a pretty frequent request, see https://musescore.org/en/node/313555.
> If so maybe change the commit title to start with "Fix #313555: "?
> 
> Isn't there any UI to switch this on/off?

@Jojo-Schmitz there is no UI, I added the code that works on MU mobile so we have the same library. I didn't know about the issue. I tested with setting in layoutoptions.h:

struct LayoutOptions
{
    LayoutMode mode = LayoutMode::HORIZONTAL_FIXED;

File: src/engraving/libmscore/measure.cpp (hunk 1 above)
Comment (alexpavlov96) [reply src/engraving/libmscore/measure.cpp:4859]: it chanegs the current segment

Comment (Jojo-Schmitz) [reply src/engraving/libmscore/measure.cpp:4859]: Yes, I see that it used the side effect, but that's confusing code, to me at least.

**Comment by Jojo-Schmitz** [comment]
OK, so a UI for this is missing, but needed.

File: src/engraving/libmscore/measure.cpp (hunk 1 above)
Comment (igorkorsukov) [reply src/engraving/libmscore/measure.cpp:4859]: I agree with JoJo, I believe needs rewrite this code

File: src/engraving/libmscore/masterscore.h [hunk 2]
```diff
@@ -66,6 +66,8 @@ class MasterScore : public Score
     QSet<int> occupiedMidiChannels;                   // each entry is port*16+channel, port range: 0-inf, channel: 0-15
     unsigned int searchMidiMappingFrom = 0;           // makes getting next free MIDI mapping faster
 
+    qreal m_widthOfSegmentCell { 3 };
```
Comment (igorkorsukov) [inline src/engraving/libmscore/masterscore.h:69]: needs `qreal m_widthOfSegmentCell = 3.0;`(https://github.com/musescore/MuseScore/wiki/CodeGuidelines#declaring-variables)

File: src/engraving/libmscore/measure.cpp [hunk 3]
```diff
@@ -4823,4 +4824,238 @@ void Measure::computeMinWidth()
...
 }
+
+void Measure::layoutSegmentsInPracticeMode(const std::vector<int>& visibleParts)
+{
+    layoutSegmentsWithDuration(visibleParts);
+}
+
+qreal Measure::computeFirstSegmentXPosition(Segment* segment)
+{
+    qreal x{ 0 };
```
Comment (igorkorsukov) [inline src/engraving/libmscore/measure.cpp:4835]: needs `qreal x = 0.0;` 
https://github.com/musescore/MuseScore/wiki/CodeGuidelines#declaring-variables

File: src/engraving/libmscore/measure.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply src/engraving/libmscore/measure.cpp:4860]: yes, that while loop is better

**Comment by alexpavlov96** [comment]
> OK, so a UI for this is not part of this PR (yet), but needed.
> At least an advanced preference setting.

@Jojo-Schmitz, no UI yet, so I didn't change the name

**Comment by igorkorsukov** [comment]
We plan to make the UI later

**Pull request merged**